deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -292,88 +281,59 @@ private TrustManagerFactory generateTrustManagerFactory(IotHubCertificateManager
+        BufferedReader br = new BufferedReader(new FileReader(filename));
+        String line;
+        while ((line = br.readLine()) != null) {
+            strKeyPEM += line + "\n";
         }
+        br.close();
+        return strKeyPEM;
     }
 
-    private static Collection<X509Certificate> parsePublicKeyCertificate(String publicKeyCertificateString) throws CertificateException
+    public static RSAPrivateKey parsePrivateKeyString(String privateKeyPEM) throws IOException, GeneralSecurityException
     {
-        try
+        if (privateKeyPEM == null || privateKeyPEM.isEmpty())
         {
-            Collection<X509Certificate> certChain = new ArrayList<>();
-
-            // Codes_SRS_IOTHUBSSLCONTEXT_34_033: [This function shall return the X509Certificate cert chain specified by the PEM formatted publicKeyCertificateString.]
-            Security.addProvider(new BouncyCastleProvider());
-
-            CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-            final PemReader publicKeyCertificateReader = new PemReader(new StringReader(publicKeyCertificateString));
-
-            try
-            {
-                PemObject possiblePublicKeyCertificate;
-                while (((possiblePublicKeyCertificate = publicKeyCertificateReader.readPemObject()) != null))
-                {
-                    byte[] content = possiblePublicKeyCertificate.getContent();
-                    if (content.length > 0)
-                    {
-                        final ByteArrayInputStream bais = new ByteArrayInputStream(content);
-
-                        while (bais.available() > 0)
-                        {
-                            final Certificate cert = certFactory.generateCertificate(bais);
-                            if (cert instanceof X509Certificate)
-                            {
-                                certChain.add((X509Certificate) cert);
-                            }
-                        }
-                    }
-                    else
-                    {
-                        break;
-                    }
-                }
-            }
-            finally
-            {
-                publicKeyCertificateReader.close();
-            }
-
-            return certChain;
-        }
-        catch (Exception e)
-        {
-            // Codes_SRS_IOTHUBSSLCONTEXT_34_034: [If any exception is encountered while attempting to create the public key certificate instance, this function shall throw a CertificateException.]
-            throw new CertificateException(e);
+            throw new GeneralSecurityException("Public key certificate cannot be null or empty");
         }
+
+        privateKeyPEM = privateKeyPEM.replace("-----BEGIN PRIVATE KEY-----\n", "");
+        privateKeyPEM = privateKeyPEM.replace("-----END PRIVATE KEY-----", "");
+        byte[] encoded = Base64.decodeBase64(privateKeyPEM.getBytes("UTF-8"));
+        KeyFactory kf = KeyFactory.getInstance("RSA");
```

Comment:
Double check if `RSA` is the only encryption algorithm allowed. What about `AES` or `ECC`?

Reply:
Currently, the SDK only supports RSA certs, so this PR isn't changing our stance on AES or ECC. I'd rather follow the approach you mention below where we just take an SSLContext from a new device client constructor. That way, customers can parse their certs however they'd like. 

That said, we cannot remove support of parsing RSA certs from our existing constructors. The tests we have in place already verified that we are safe making this change from bouncycastle to a simpler parser.